.github/workflows/pr-preview.yaml

```diff
@@ -0,0 +1,203 @@
+name: Deploy PR Preview
+
+on:
+  pull_request:
+    types: [opened, synchronize, reopened, closed]
+
+# Cancel previous runs when new commits are pushed to the PR
+concurrency:
+  group: pr-preview-${{ github.event.pull_request.number }}
+  cancel-in-progress: true
+
+jobs:
+  deploy-preview:
+    name: Deploy PR Preview
+    runs-on: ubuntu-latest
+    if: github.event.action != 'closed'
+    permissions:
+      pull-requests: write
+      deployments: write
+      contents: read
+
+    steps:
+      - name: Checkout PR code
+        uses: actions/checkout@v6
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+
+      - name: Setup Node.js
+        uses: actions/setup-node@v6
+        with:
+          node-version: '22'
+          cache: 'npm'
+          cache-dependency-path: 'package-lock.json'
+
+      - name: Install dependencies
+        run: |
+          echo "Installing dependencies..."
+          npm ci
+
+      - name: Build Docusaurus site
+        env:
+          DOCUSAURUS_BASE_URL: /pr-${{ github.event.pull_request.number }}
+        run: |
+          echo "Building site with base URL: $DOCUSAURUS_BASE_URL"
+          npm run build
+
+      - name: Upload build artifact for local testing
+        uses: actions/upload-artifact@v4
+        with:
+          name: pr-${{ github.event.pull_request.number }}-build
+          path: build/
+          retention-days: 30
+
+      - name: Prepare deployment directory
+        run: |
+          PR_DIR="pr-${{ github.event.pull_request.number }}"
+          echo "Creating directory: $PR_DIR"
+          mkdir -p "$PR_DIR"
+
+          # Create config.yaml
+          cat > "$PR_DIR/config.yaml" << 'EOF'
+          static:
+            files: 'build/**'
+            urlPath: 'pr-${{ github.event.pull_request.number }}'
+            extensions: ['html']
+            index: true
+          EOF
+
+          # Copy build directory
+          cp -r build "$PR_DIR/"
+
+          echo "Contents of $PR_DIR:"
+          ls -la "$PR_DIR"
+
+      - name: Install HarperDB CLI
+        run: |
+          npm install harperdb
```

Comment:
Why not install harper globally? Then I would think you wouldn't need `npx` on the next step, right?